### PR TITLE
Code sample from file

### PR DESCRIPTION
**Pull request by davifantasia**
Closes https://github.com/commercetools/commercetools-docs-kit/issues/64

### CODE REVIEW

File: packages/gatsby-theme-docs/gatsby-node.js [hunk 1]
```diff
@@ -209,6 +210,18 @@ exports.onCreateWebpackConfig = ({ actions, getConfig }) => {
         },
       ],
     },
+    {
+      test: /\.(js|java)$/,
```
Comment (emmenko) [inline packages/gatsby-theme-docs/gatsby-node.js]: This shouldn't be hardcoded

File: packages/gatsby-theme-docs/src/components/code-sample-from-file.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { CodeBlock } from '@commercetools-docs/ui-kit';
+
+// import Java from '../code-samples/test.java';
+
+// CodeBlockFromImport
+function CodeSampleFromFile(props) {
+  const [sample, setSample] = useState('');
+
+  // props.import - path to code sample
+  import(`../code-samples/${props.name}`).then(result => {
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/code-sample-from-file.js]: How does this work with SSR?

File: packages/gatsby-theme-docs/src/components/code-sample-from-file.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    setSample(result.default);
+  });
+
+  return <CodeBlock content={sample} language={props.language} />;
+}
+
+CodeSampleFromFile.propTypes = {
+  name: PropTypes.string,
+  language: PropTypes.string,
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/code-sample-from-file.js]: The language should be derived by the file name extension, we don't need to pass it explicitly.

File: packages/gatsby-theme-docs/src/loaders/code-sample.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+function codeSample(content) {
+  this.cacheable();
+
+  this.value = content;
+  return `module.exports = ${JSON.stringify(content)}`;
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/loaders/code-sample.js]: So basically you're wrapping the file content into an export.

What type is the `content`? A string or what?

File: packages/gatsby-theme-docs/src/components/code-sample-from-file.js (hunk 2 above)
Comment (emmenko) [reply packages/gatsby-theme-docs/src/components/code-sample-from-file.js]: Why not using a `require`?

**Comment by davifantasia** [comment]
Closing this in favour of https://github.com/commercetools/commercetools-docs-kit/pull/300

**Pull request closed without merging**